PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ['digest.digest']
-    _inherit = ['discuss.channel.member']
-    _inherit = ['mail.message']
-    _inherit = ["rating.rating"]
-    _inherit = ['res.users.settings']

Total Changes: 16

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/im_livechat/models/chatbot_script.py
+++ b/addons/im_livechat/models/chatbot_script.py
@@ -7,6 +7,7 @@ from odoo.tools import email_normalize, get_lang, html2plaintext, is_html_empty,
 
 
 class ChatbotScript(models.Model):
+    _name = 'chatbot.script'
     _description = 'Chatbot Script'
     _inherit = ['image.mixin', 'utm.source.mixin']
     _rec_name = 'title'

--- a/addons/im_livechat/models/chatbot_script_answer.py
+++ b/addons/im_livechat/models/chatbot_script_answer.py
@@ -9,6 +9,7 @@ import textwrap
 
 
 class ChatbotScriptAnswer(models.Model):
+    _name = 'chatbot.script.answer'
     _description = 'Chatbot Script Answer'
     _order = 'script_step_id, sequence, id'
 

--- a/addons/im_livechat/models/chatbot_script_step.py
+++ b/addons/im_livechat/models/chatbot_script_step.py
@@ -13,6 +13,7 @@ from markupsafe import Markup
 
 
 class ChatbotScriptStep(models.Model):
+    _name = 'chatbot.script.step'
     _description = 'Chatbot Script Step'
     _order = 'sequence, id'
     _rec_name = 'message'

--- a/addons/im_livechat/models/digest.py
+++ b/addons/im_livechat/models/digest.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class DigestDigest(models.Model):
-    _inherit = ['digest.digest']
+    _inherit = 'digest.digest'
 
     kpi_livechat_rating = fields.Boolean('% of Happiness')
     kpi_livechat_rating_value = fields.Float(digits=(16, 2), compute='_compute_kpi_livechat_rating_value')

--- a/addons/im_livechat/models/discuss_channel_member.py
+++ b/addons/im_livechat/models/discuss_channel_member.py
@@ -7,7 +7,7 @@ from odoo.addons.mail.tools.discuss import Store
 
 
 class DiscussChannelMember(models.Model):
-    _inherit = ['discuss.channel.member']
+    _inherit = 'discuss.channel.member'
 
     @api.autovacuum
     def _gc_unpin_livechat_sessions(self):

--- a/addons/im_livechat/models/ir_binary.py
+++ b/addons/im_livechat/models/ir_binary.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class IrBinary(models.AbstractModel):
-    _inherit = ["ir.binary"]
+    _inherit = "ir.binary"
 
     def _find_record_check_access(self, record, access_token, field):
         """Custom access check allowing to retrieve an operator's avatar.

--- a/addons/im_livechat/models/mail_message.py
+++ b/addons/im_livechat/models/mail_message.py
@@ -5,7 +5,7 @@ from odoo.addons.mail.tools.discuss import Store
 
 
 class MailMessage(models.Model):
-    _inherit = ['mail.message']
+    _inherit = 'mail.message'
 
     parent_author_name = fields.Char(compute="_compute_parent_author_name")
     parent_body = fields.Html(compute="_compute_parent_body")

--- a/addons/im_livechat/models/rating_rating.py
+++ b/addons/im_livechat/models/rating_rating.py
@@ -5,8 +5,7 @@ from odoo import api, models
 
 
 class RatingRating(models.Model):
-
-    _inherit = ["rating.rating"]
+    _inherit = "rating.rating"
 
     @api.depends('res_model', 'res_id')
     def _compute_res_name(self):

--- a/addons/im_livechat/models/res_users_settings.py
+++ b/addons/im_livechat/models/res_users_settings.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResUsersSettings(models.Model):
-    _inherit = ['res.users.settings']
+    _inherit = 'res.users.settings'
 
     livechat_username = fields.Char("Livechat Username", help="This username will be used as your name in the livechat channels.")
     livechat_lang_ids = fields.Many2many(comodel_name='res.lang', string='Livechat languages',
